x/emissions/proto/emissions/v3/genesis.proto

```diff
@@ -0,0 +1,373 @@
+syntax = "proto3";
+package emissions.v3;
+
+import "amino/amino.proto";
+import "cosmos_proto/cosmos.proto";
+import "emissions/v3/node.proto";
+import "emissions/v3/nonce.proto";
+import "emissions/v3/params.proto";
+import "emissions/v3/reputer.proto";
+import "emissions/v3/score.proto";
+import "emissions/v3/stake.proto";
+import "emissions/v3/topic.proto";
+import "emissions/v3/types.proto";
+import "emissions/v3/worker.proto";
+import "gogoproto/gogo.proto";
+
+option go_package = "github.com/allora-network/allora-chain/x/emissions/types";
+
+// GenesisState is the state that must be provided at genesis.
+message GenesisState {
+  reserved 13, 14, 15;
+  reserved "latestInfererScoresByWorker", "latestForecasterScoresByWorker", "latestReputerScoresByReputer";
+
+  // params defines all the parameters of the module.
+  Params params = 1 [(gogoproto.nullable) = false];
+
+  /// TOPIC
+  // the next topic id to be used, equal to the number of topics that have been
+  // created
+  uint64 nextTopicId = 3;
+  // every topic that has been created indexed by their topicId starting from 1
+  // (0 is reserved for the root network)
+  repeated TopicIdAndTopic topics = 4;
+  repeated uint64 activeTopics = 5;
+  // every topic that has been churned and ready to be rewarded i.e. reputer
+  // losses have been committed
+  repeated uint64 rewardableTopics = 6;
+  // for a topic, what is every worker node that has registered to it?
+  repeated TopicAndActorId topicWorkers = 7;
+  // for a topic, what is every reputer node that has registered to it?
+  repeated TopicAndActorId topicReputers = 8;
+  // map of (topic) -> nonce/block height
+  repeated TopicIdAndBlockHeight topicRewardNonce = 9;
+
+  /// SCORES
+  // map of (topic, block_height, worker) -> score
+  repeated TopicIdBlockHeightScores infererScoresByBlock = 10;
+  // map of (topic, block_height, worker) -> score
+  repeated TopicIdBlockHeightScores forecasterScoresByBlock = 11;
+  // map of (topic, block_height, reputer) -> score
+  repeated TopicIdBlockHeightScores reputerScoresByBlock = 12;
+  // map of (topic, block_height, worker) -> score
+  repeated TopicIdActorIdScore infererScoreEmas = 60;
+  // map of (topic, block_height, worker) -> score
+  repeated TopicIdActorIdScore forecasterScoreEmas = 61;
+  // map of (topic, block_height, reputer) -> score
+  repeated TopicIdActorIdScore reputerScoreEmas = 62;
```

Comment:
yes I wrote that code, is there a problem?

I thought it would be better to not reuse fields because our existing testnet genesis calls them latestBlah 

this way they get put into a new json field?

Or was I wrong

Reply:
If you use the same tag and changes the type it remains the same field with a different type, which can mess with deserialization. The thing is that you also changed the tag number, so maybe it will be a different field. Have to check.

Reply:
Actually, there is no issue. The name is also different.